### ISSUE TITLE
Can't bind ONNXRT C++ Application to single core using numa with sequential executor mode using default CPU EP (MLAS)

### ISSUE THREAD

**Issue by MaajidKhan** [microsoft/onnxruntime#10377]
**Describe the bug**
The onnxruntime_perf_test c++ application has two different modes of execution - sequential executor mode (default) and the parallel executor mode. By default, the application tries to exercise all the cores of the machine in both modes. However, we wanted to see the performance using single core and we used numa to bind the application to single core. Our observation was the following:

**1. With Sequential executor mode** - Though we use numa to bind the application to 1 core, it doesn't work and it ends up working as default (where it uses all cores)

**2. With Parallel executor mode** - When we use numa to bind the application to 1 core, it works with this mode and the application runs on single core.

**Urgency**
Medium.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  **Ubuntu 20.04.3 LTS**
- ONNX Runtime installed from (source or binary): **source**
- ONNX Runtime version: **v1.10.0**
- Python version: **3.8.10**
- Visual Studio version (if applicable): NA
- GCC/Compiler version (if compiling from source): **7.5.0**
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**To Reproduce**
- **Describe steps/code to reproduce the behavior.**
Build default CPU EP (MLAS) with ONNXRT v1.10.0 release.

- **Attach the ONNX model to the issue (where applicable) to expedite investigation.**
Download the tiny_yolov2 opset8 ONNX model from here:
https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/tiny-yolov2/model

**Expected behavior**
When the c++ applications is ran using sequential executor mode and if numa is used to bind it to single core, it should work as expected.

**Screenshots**

**MLAS -Sequential Executor mode:**
![image](https://user-images.githubusercontent.com/29036417/150929496-2f476765-0e55-476c-b577-3935872f79b7.png)

**MLAS - Parallel Executor Mode:**
![image](https://user-images.githubusercontent.com/29036417/150929742-60deafb3-3921-4ec4-87fc-485b06cf8c47.png)

**Note:**
we used Intel's VTune profiler Tool to get these screenshots of the core usage stats. The machine we used has 16 cores.

**Machine configuration:**
CPU(s):                          32
On-line CPU(s) list:             0-31
Thread(s) per core:              2
Core(s) per socket:              16

**Additional context**
Add any other context about the problem here. If the issue is about a particular model, please share the model details as well to facilitate debugging.

**Note:** This issue is seen only with default CPU EP. However we didn't observe similar issue while using OpenVINO EP.
so, also attaching the commands for OpenVINO EP as well, if you would like to try this out as well.

**commands to reproduce the same:**

**case 1: sequential executor mode with all cores (default case)**

**OpenVINO EP:**
./onnxruntime_perf_test  -r 1 -c 1 -v -e openvino model.onnx

**MLAS:**
./onnxruntime_perf_test  -r 1 -c 1 -v -e cpu model.onnx

**case 2: sequential executor mode with single core using numa**

**OpenVINO EP:**
numactl -m 0 --physcpubind 0 ./onnxruntime_perf_test  -r 1 -c 1 -v -e openvino model.onnx

**MLAS:**
numactl -m 0 --physcpubind 0 ./onnxruntime_perf_test  -r 1 -c 1 -v -e cpu model.onnx

**case 3: Parallel executor mode with all cores**

**OpenVINO EP:**
./onnxruntime_perf_test  -r 1 -c 1 -P -v -e openvino model.onnx

**MLAS:**
./onnxruntime_perf_test  -r 1 -c 1 -P -v -e cpu model.onnx

**case 4: Parallel executor mode with single core using numa  (works)**

**OpenVINO EP:**
numactl -m 0 --physcpubind 0 ./onnxruntime_perf_test  -r 1 -c 1 -P -v -e openvino model.onnx

**MLAS:**
numactl -m 0 --physcpubind 0 ./onnxruntime_perf_test  -r 1 -c 1 -P -v -e cpu model.onnx

**Comment by jywu-msft** [comment]
@snnn is this related to https://github.com/microsoft/onnxruntime/issues/7347 which is assigned to you?

**Comment by snnn** [comment]
No, it is different.

**Comment by snnn** [comment]
See https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/test/perftest/ort_test_session.cc#L408 . Please set the value to one. But in general I don't like it. I don't think we can get a fair result of single thread perf in the case that  you will give all the CPU cache to a single core. You will get a better than real number.

**Comment by MaajidKhan** [comment]
Thanks @snnn. We tried out the above option as mentioned and it works.
My understanding from my experiments is that this config - "intra_op_num_threads" sets number of threads (virtual cores) and not the actual core itself.

I was able to use -x 1 with sequential_executor mode with MLAS using onnxruntime_perf_test and it did only use 1 virtual core. 

Though, this option really helps us in controlling the usage of number of virtual cores, but really doesn't give us more control on to decide which exact virtual core I would like my application to use. This is where numa usually helps.
